src/main/java/games/strategy/triplea/delegate/BattleTracker.java

```diff
@@ -1135,6 +1137,36 @@ void fightAirRaidsAndStrategicBombing(final IDelegateBridge delegateBridge, Supp
     }
   }
 
+  /**
+   * 'Auto-fight' defenseless battles.
+   */
+  public void fightAutoKills(final IDelegateBridge bridge) {
+    // Kill undefended transports. Done first to remove potentially dependent sea battles
+    // Which could block amphibious assaults below
+    final GameData gameData = bridge.getData();
+    for (final Territory territory : getPendingBattleSites(false)) {
+      final IBattle battle = getPendingBattle(territory, false, BattleType.NORMAL);
+      final List<Unit> defenders = new ArrayList<>();
+      defenders.addAll(battle.getDefendingUnits());
```

Comment:
It is necessary to convert from the Collection returned by battle.getDefendingUnits() into a List needed by one or both of the following calls. I'm open to suggestions on a better way of doing it though.